openeo/extra/job_management.py

```diff
@@ -30,6 +31,22 @@ class _Backend(NamedTuple):
 
 MAX_RETRIES = 5
 
+
+class JobDatabaseInterface:
+
+    def exists(self) -> bool:
+        """Is there existing data in the database?"""
+        return False
+
+    def read(self) -> pd.DataFrame:
+        """Read job data from the database ad pandas DataFrame."""
+        raise NotImplementedError()
+
+    def persist(self, df: pd.DataFrame):
+        """Store job data to the database."""
+        raise NotImplementedError()
```

Comment:
Is there a reason not to make this a formal interface? As a user it might be beneficial to immediately get thrown an error if your custom class doesn't have all necessary methods.

Reply:
Indeed, now that we are making this interface more public, that would be useful